# **UFO Sightings - Part 1: Preprocessing Data**

## 0. Table of Contents

[1. Importing Data Set and Necessary Packages](#1-importing-data-set-and-necessary-packages)<br>
[2. Data Cleaning](#2-data-cleaning)

In [20]:
import re

# Definieer een lijst met je sectietitels
headers = ["1. Importing Data Set and Necessary Packages", "2. Data Cleaning"]

# Maak Markdown-links
for header in headers:
    link = "#" + re.sub(r"[^a-zA-Z0-9 ]", "", header).lower().replace(" ", "-")
    print(f"- [{header}]({link})")


- [1. Importing Data Set and Necessary Packages](#1-importing-data-set-and-necessary-packages)
- [2. Data Cleaning](#2-data-cleaning)


## 1. Importing Data Set and Necessary Packages

In [13]:
import html                                  # package used to convert text
import numpy as np                           # linear algebra
import os                                    # package used to set work directory
import pandas as pd                          # package used to create dataframes
import re                                    # regular expressions

from datetime import datetime                # package used to convert string to datetime 

In [16]:
data = pd.read_csv('scrubbed.csv', low_memory=False)

In [14]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80332 entries, 0 to 80331
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   datetime              80332 non-null  object 
 1   city                  80332 non-null  object 
 2   state                 74535 non-null  object 
 3   country               70662 non-null  object 
 4   shape                 78400 non-null  object 
 5   duration (seconds)    80332 non-null  object 
 6   duration (hours/min)  80332 non-null  object 
 7   comments              80317 non-null  object 
 8   date posted           80332 non-null  object 
 9   latitude              80332 non-null  object 
 10  longitude             80332 non-null  float64
dtypes: float64(1), object(10)
memory usage: 6.7+ MB
None


The data set contains a decent amount of observations, 80,332 to be specific, and 11 columns. Apart from the column `longitude`, the columns are not assigned a data type. Before we assign data types, we will first do some further exploration. The columns `state`, `country`, `shape` and `comments` seem to have missing values which should be addressed as well. There are two different columns that contain the durations of the UFO sightings. All of these points will be addressed in the next section where we will be cleaning and preparing the data for analysis.

## 2. Data Cleaning

*coming soon*